library/platform_util.c

```diff
@@ -168,13 +168,34 @@ uint32_t mbedtls_platform_random_in_range( size_t num )
 #endif
 }
 
-/* Some compilers (armcc 5 for example) optimize away successive reads from a
- * volatile local variable (which we use as a counter-measure to fault
- * injection attacks), unless there is a call to an external function between
- * them. This functions doesn't need to do anything, it just needs to be
- * in another compilation unit. So here's a function that does nothing. */
+int mbedtls_platform_random_delay( size_t max_rand )
+{
+#if !defined(MBEDTLS_ENTROPY_HARDWARE_ALT)
+    (void) max_rand;
+    return -1;
+#else
+    size_t random_number;
+    volatile size_t i = 0;
+    if( max_rand == 0 || max_rand > INT_MAX )
+    {
+        return( -1 );
+    }
+
+    random_number = mbedtls_platform_random_in_range( max_rand );
+
+    do
+    {
+        i++;
+    } while( i < random_number );
+
+    return( (int)i );
+#endif /* !MBEDTLS_ENTROPY_HARDWARE_ALT */
+}
+
 void mbedtls_platform_enforce_volatile_reads( void )
```

Comment:
Minor: The function name made sense when it was a dummy function, but now it's actually doing something every time in providing a random delay. 

Given it's simply calling a function of `mbedtls_platform_random_delay`, I'm not sure we need both functions, and together they don't make sense.